StyleCop.Analyzers/StyleCop.Analyzers.CodeGeneration/SyntaxLightupGenerator.cs

```diff
@@ -1393,5 +1441,20 @@ public string GetAccessorResultType(SyntaxData syntaxData)
                 return null;
             }
         }
+
+        private record CompilationData(ImmutableDictionary<string, ExistingTypeData> ExistingTypes)
```

Comment:
@chsienki is there a straightforward way to have a properly equatable dictionary in a record, without having to implement custom equality?